net/tstun/wrap.go

```diff
@@ -711,8 +711,7 @@ func peerConfigTableFromWGConfig(wcfg *wgcfg.Config) *peerConfigTable {
 	// require masquerading.
 	exitNodeRequiresMasq := false // true if using an exit node and it requires masquerading
 	for _, p := range wcfg.Peers {
-		isExitNode := slices.Contains(p.AllowedIPs, tsaddr.AllIPv4()) || slices.Contains(p.AllowedIPs, tsaddr.AllIPv6())
-		if isExitNode {
+		if tsaddr.ContainsEitherExitRoutes(views.SliceOf(p.AllowedIPs)) {
```

Comment:
This was previously checking for 0/0, while now will match any /0 route. I'd suggest to return the previous behaviour.